### ISSUE TITLE
Word 2010/2013: when navigating list of font names and sizes, NVDA does not announce each option

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#4992]
**Reported by nvdakor on 2015-03-23 04:10**
Hi,
A member of the NVDA users list reports:
When you navigate through font names and sizes, NVDA does not announce each option (confirmed).
STR:
1. While in a document in Word 2010/2013, open Home ribbon.
2. Press TAB until you arrive at font name and font size edit fields.
3. Press Alt+Down arrow to open font name/size options.
Expected: NVDA announces each option (font name or size).
Actual: NVDA does not announce anything.

Technical: Mick, could you verify that we're working with yet another broken UIA implementation? The dev info for one of the font name options is:

```
Developer info for navigator object:
name: u''
role: ROLE_POPUPMENU
states: STATE_FOCUSABLE
isFocusable: True
hasFocus: 0
Python object: <NVDAObjects.UIA.UIA object at 0x04C38810>
Python class mro: (<class 'NVDAObjects.UIA.UIA'>, <class 'NVDAObjects.window.Window'>, <class 'NVDAObjects.NVDAObject'>, <class 'baseObject.ScriptableObject'>, <class 'baseObject.AutoPropertyObject'>, <type 'object'>)
description: u''
location: (162, 84, 285, 539)
value: None
appModule: <'appModuleHandler' (appName u'winword', process ID 2340) at address 4cb1170>
appModule.productName: u'Microsoft Office 2013'
appModule.productVersion: u'15.0.4701.1001'
TextInfo: <class 'NVDAObjects.NVDAObjectTextInfo'>
windowHandle: 264736
windowClassName: u'Net UI Tool Window'
windowControlID: 0
windowStyle: -1778384896
windowThreadID: 2024
windowText: u''
displayText: u'Theme Fonts\nRecently Used FontsRecently Used FontsRecently Used Fonts\nAll Fonts Used FontsAll Fonts'
UIAElement: <POINTER(IUIAutomationElement) ptr=0x61dc2e0 at 4c46df0>
UIA automationID: 
UIA frameworkID: Win32
UIA runtimeID: (42, 264736)
UIA providerDescription: [Main:Nested [pid:2340,hwnd:0x40A20 Main(parent link):Unidentified Provider (unmanaged:mso.dll)](pid:3772,hwnd:0x40A20); Hwnd(parent link):Microsoft: HWND Proxy (unmanaged:UIAutomationCore.dll)]
UIA className: Net UI Tool Window
```

At least I got this when I press NVDA+F1 after arrowing through the font name menu.

Setup:
- Reporter: Word 2013.
- Me: Word 2013 on Windows 8.1.
  Thanks.

**Comment by bhavyashah** [comment]
I suspect this ticket has a duplicate somewhere, although I might well be wrong.
@Qchristensen @josephsl Could you please retest this issue?

**Comment by Qchristensen** [comment]
I actually don't have either Office 2010 or 2013 handy, but testing in 2007 and 2016 which I do have handy, I found that alt+down arrow doesn't actually open the drop down list, it moves to the end of the text currently in the selected drop down list and unselected the text (which results in the current font being read again as unselected).  Pressing down arrow on its own does read each font name or font size as I move through them.  I do see an debug warning in the log for each keypress here though that doesn't seem to be interfering with the reading for me (NVDA 2017.2 and Word 2016 for this):

"IO - inputCore.InputManager.executeGesture (18:35:00):
Input: kb(laptop):downArrow
IO - speech.speak (18:35:00):
Speaking [u'down arrow']
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (18:35:00):
oleacc.AccessibleObjectFromEvent with window 2165312, objectID 833 and childID 0: [Error -2147467259] Unspecified error
IO - speech.speak (18:35:00):
Speaking [u'Algerian']
IO - speech.speak (18:35:00):
Speaking [u'selected Algerian']
IO - inputCore.InputManager.executeGesture (18:35:01):
Input: kb(laptop):downArrow
"

It would be interesting to see if someone with Office 2010 or 2013 can reproduce the original issue?

**Comment by bhavyashah** [comment]
Testing precisely with the STR provided in the ticket body with Word 2010 and NVDA 2017.3 on a Windows 8.1 system, I was unable to reproduce the reported issue. Arrowing in both the Font size and Font (name) boxes caused NVDA to speak every option. However, one possibly related but much less significant issue I experienced while arrowing through the Font (name) box can be explained by the following log snippet:
IO - inputCore.InputManager.executeGesture (12:24:14.405):
Input: kb(laptop):upArrow
DEBUG - editableText.EditableText._hasCaretMoved (12:24:14.655):
Caret didn't move before timeout. Elapsed: 100 ms
IO - speech.speak (12:24:14.671):
Speaking [LangChangeCommand ('en_GB'), u'Arial']
IO - speech.speak (12:24:14.703):
Speaking [LangChangeCommand ('en_GB'), u'selected Arial']
Simply put, NVDA reads every option as "{font name} selected {font name}" instead of simply "{font name}".
Since @josephsl was successful in replicating the original buggy behaviour at a point in time, Joseph, could you please update us about the present standing of this issue by testing at your end?

**Comment by josephsl** [comment]
Hi,

Safe to close, as I cannot reproduce this anymore in Word 2016. In case someone does come forward and say this isn't working in 2010/2013, please reopen. Thanks.
